src/branchandbound.jl

```diff
@@ -0,0 +1,81 @@
+using TaylorModels
+# univariate
+function enclose_BranchandBound(f::Function, dom::Interval{T}; order::Int = 10,
+                                tol::Number = 0.6) where{T}
+    x0 = Interval(mid(dom))
+    x = TaylorModel1(order, x0, dom)
+    return branchandbound(f(x-x0).pol, dom, tol)
+end
+# multivariate
+function enclose_BranchandBound(f::Function, dom::IntervalBox{T,N}; order::Int = 10,
+                                tol::Number = 0.6) where {T,N}
+    n = length(dom)
+    x0 = mid(dom)
+    set_variables(Float64, "x", order=2order, numvars=n)
+    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:n]
+    return branchandbound(f(x...).pol, dom - x0, tol)
+end
+
+function branchandbound(p::Taylor1{T}, dom::Interval{W}, ϵ::Number) where{T,W}
+    return _branchandbound(p, dom, ϵ)
+end
+
+function branchandbound(p::TaylorN{T}, dom::IntervalBox{N,W}, ϵ::Number) where{T,N,W}
+    return _branchandbound(p, dom, ϵ)
+end
+
+function _branchandbound(p::Union{TaylorN{T},Taylor1{T}},
+                       dom::Union{IntervalBox{N,W},Interval{W}}, ϵ::Number) where {N,T,W}
+    K = 1
+    Rperv = evaluate(p, dom)
+    D1, D2 = bisect(dom)
+    D = [D1, D2]
```

Comment:
why not use `D = bisect(dom)` ?

Reply:
Then D will be Tuple `Tuple{IntervalBox{2,Float64},IntervalBox{2,Float64}}`  But D should be `2-element Array{IntervalBox{2,Float64},1}`

Reply:
true, but this doesn't answer my question -- where is the restriction to use a vector instead of a tuple?

Reply:
Yes, but vector is also fine, what do you think?

Reply:
if there is no reason to allocate a new vector, i would not allocate a new vector :)